pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/results/DistinctResultsBlock.java

```diff
@@ -60,23 +61,31 @@ public DataSchema getDataSchema(QueryContext queryContext) {
 
   @Override
   public Collection<Object[]> getRows(QueryContext queryContext) {
-    List<Object[]> rows = new ArrayList<>(_distinctTable.size());
-    for (Record record : _distinctTable.getRecords()) {
-      rows.add(record.getValues());
+    if (queryContext.isServerReturnFinalResult()) {
```

Comment:
i felt like there's 2 bug here @Jackie-Jiang 
1. this _distinctTable inside the DistinctResultBlock can somehow be a non-main-table..
2. currently it is returning the final Schema in the `getDataSchema` API, i suspect it is not being used on the distinct codepath but it should really return the distinct object data schema as encoded in the get datatable codepath

Reply:
bug item 2 is fixed.

Reply:
fixed. chatted with @Jackie-Jiang offline